chain/client/src/client_actor.rs

```diff
@@ -1381,8 +1383,25 @@ impl ClientActorInner {
         };
 
         // If we produced the optimistic block, send it out before we save it.
+        let epoch_id = self
+            .client
+            .chain
+            .epoch_manager
```

Comment:
nit: I guess we can add and additional line to get the epoch manager? `let epoch_manager = &self.client.chain.epoch_manager`?